client/memory.c

```diff
@@ -69,7 +69,7 @@ void xdag_mem_tempfile_path(const char *tempfile_path)
 	if (!tempfile_path)
 		return;
 
-	strncpy(g_tmpfile_path, tempfile_path, PATH_MAX);
+	strncpy(g_tmpfile_path, tempfile_path, PATH_MAX-1);
```

Comment:
why -1?
You should, instead, add \0 to the end of g_tmpfile_path to make it safe